### PR TITLE
Add concept about envelope encryption

### PR DESCRIPTION
**Pull request by osg-grafana**
Partially fixes https://github.com/grafana/grafana-enterprise/issues/2196.

### CODE REVIEW

**Review by mjseaman — APPROVED** [review]

**Comment by osg-grafana** [comment]
Please use suggestions instead of direct commits.

File: docs/sources/administration/envelope-encryption.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+In Grafana, you can choose to use envelope encryption. Instead of
+encrypting all secrets with a single key, Grafana uses a set of keys
+called data encryption keys (DEKs) to encrypt them. These data
+encryption keys are themselves encrypted with a single key encryption
+key (KEK).
+
+To turn on envelope encryption, add the term `envelopeEncryption` to the list of feature toggles in your [Grafana configuration]({{< relref "../administration/configuration/#feature_toggles" >}}).
+
+**> Note:** Back up your database before turning on envelope encryption for the first time. After you have turned envelope encryption on, avoid turning it off. If you turn envelope encryption on, create new secrets or update
+your existing secrets (for example, by creating a new data source or alert notification channel), and then turn envelope encryption off, then those data sources, alert notification channels, and other resources that are using envelope encryption will stop working and you will experience errors. This is because the secrets that are encrypted with envelope encryption cannot be decrypted or used by Grafana when envelope encryption is turned off.
```
Comment (osg-grafana) [inline docs/sources/administration/envelope-encryption.md:21]: @mjseaman Can you take over this PR and reword this. Per our DM, this is a bit large to parse. Can we break down this complexity? In the meantime, I will continue working on the other related PRs that will not cause merge conflicts with this one.

Comment (mjseaman) [reply docs/sources/administration/envelope-encryption.md]: @osg-grafana I gave it an update; lmk what you think. I'm not sure if there is a way to put process steps in a quotation block?

File: docs/sources/administration/envelope-encryption.md [hunk 2]
```diff
@@ -17,5 +17,9 @@ key (KEK).
...
 
-**> Note:** Back up your database before turning on envelope encryption for the first time. After you have turned envelope encryption on, avoid turning it off. If you turn envelope encryption on, create new secrets or update
-your existing secrets (for example, by creating a new data source or alert notification channel), and then turn envelope encryption off, then those data sources, alert notification channels, and other resources that are using envelope encryption will stop working and you will experience errors. This is because the secrets that are encrypted with envelope encryption cannot be decrypted or used by Grafana when envelope encryption is turned off.
+**> Note:** Back up your database before turning on envelope encryption for the first time. After you have turned envelope encryption on, avoid turning it off. You will experience errors if you:
+1. Turn envelope encryption on.
+2. Create new secrets or update your existing secrets (for example, by creating a new data source or alert notification channel).
+3. Turn envelope encryption off.
```
Comment (osg-grafana) [inline docs/sources/administration/envelope-encryption.md:23]: heh, yes, you are correct. This is worse. Now I want to perform these steps.

Comment (osg-grafana) [reply docs/sources/administration/envelope-encryption.md:23]: Let's revert this, go with your comma suggestion, and squash and merge for now.

**Review by mjseaman — APPROVED** [review]

**Pull request merged**